hedera-node/docs/design/services/smart-contract-service/evm-transaction-response-codes.md

```diff
@@ -0,0 +1,59 @@
+# EVM Transaction Response Codes in Hedera
+
+When executing an EVM transaction to a smart contract using `contractCall` in Hedera and it fails, a response code is present in the transaction receipt to indicate the top-level status.
+
+The default status for a failing `contractCall` is CONTRACT_REVERT_EXECUTED, which is included in the receipt. Child transactions might have more specific errors.
+In certain cases, the top-level status may differ from CONTRACT_REVERT_EXECUTED. The errors listed below are those special cases that can occur during the execution of a transaction and will be propagated as the top-level status.
+
+The handling of those cases currently happens [here](https://github.com/hashgraph/hedera-services/blob/774ed309600e4b4acd9e1ca72fcd87d354c8b9ff/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/hevm/HederaEvmTransactionResult.java#L141-L169), in the `HederaEvmTransactionResult` in the `finalStatus()` method.
+
+
+## Possible Propagated Response Codes
+
+Note: The description for the statuses is copied from [protobuf repo](https://github.com/hashgraph/hedera-protobufs/blob/main/services/response_code.proto)
+
+- **INVALID_SOLIDITY_ADDRESS**
+    -  The system is not able to find the user’s Solidity address.
+
+- **INVALID_ALIAS_KEY**
+    -  An alias used in a CryptoTransfer transaction is not the serialization of a primitive Key message. This means a Key with a single Ed25519 or ECDSA(secp256k1) public key and no unknown protobuf fields.
+
+- **INVALID_SIGNATURE**
+    -  The transaction signature is not valid.
+
+- **MAX_ENTITIES_IN_PRICE_REGIME_HAVE_BEEN_CREATED**
```

Comment:
Yikes!  Might (maybe) be worth segregating - or marking with an asterisk - a couple of these which are real anomalies - nothing you'd actually expect to get.  This particular one indicates we've got too many total contracts in the entire system (I think).  If someone hits this, well, there's nothing they can do about it except wait until the network is fixed.

Similar for `MAX_STORAGE_IN_PRICE_REGIME_HAS_BEEN_USED`.  

`MAX_CONTRACT_STORAGE_EXCEEDED` is almost but not quite the same - that does apply to an individual contract, not the entire system. The configured limit is huge and no contract is likely to reach it in normal operation anytime soon, but you can hit it you really apply yourself.

Reply:
Yes, agreed. I've included those in a "separate" section and added some clarifying comments.